com.unity.netcode.gameobjects/Runtime/NetworkVariable/NetworkVariable.cs

```diff
@@ -83,11 +83,6 @@ public virtual T Value
 
         private protected void Set(T value)
         {
-            if (EqualityComparer<T>.Default.Equals(m_InternalValue, value))
-            {
-                return;
-            }
-
             m_IsDirty = true;
```

Comment:
so, from now on, we're effectively marking the variable dirty regardless of whether it's changed or not automatically — which makes sense as you said, snapshotting would be a better way to handle deltas.